libqtile/widget/currentscreen.py

```diff
@@ -45,15 +45,11 @@ def _configure(self, qtile, bar):
 
     def setup_hooks(self):
         def hook_response():
-            self.update_text()
-            self.bar.draw()
-
-        hook.subscribe.current_screen_change(hook_response)
-
-    def update_text(self):
-        if self.qtile.current_screen == self.bar.screen:
-            self.text = self.active_text
-            self.foreground = self.active_color
-        else:
-            self.text = self.inactive_text
-            self.foreground = self.inactive_color
+            if self.qtile.current_screen == self.bar.screen:
+                self.foreground = self.active_color
+                self.update(self.active_text)
+            else:
+                self.foreground = self.inactive_color
+                self.update(self.inactive_text)
+
+        hook.subscribe.current_screen_change(self.hook_response)
```

Comment:
Is there any reason `hook_response` is dynamically constructed within `setup_hooks` and not a class method? Looks like you hooked `self.hook_response` as if it was.

Reply:
Oh, I think I had it that way before and didn't fix this. I can make it a class method, thanks for catching this.